### PR TITLE
Update README.md to add missing Requirements section

### PR DESCRIPTION
**Pull request by manez**
# What does this Pull Request do?

Adds the missing Requirements section to bring it in line with the readme template. Took verbiage from the wiki: https://wiki.duraspace.org/display/ISLANDORA/Compound+Solution+Pack

# How should this be tested?

Have a look at the changes, verify the links work and that these are the actual requirements. Check against [readme template](https://github.com/Islandora/islandora/wiki/Readme-template).

# Interested parties
@DiegoPino or maintainer @whikloj

### CODE REVIEW

**Comment by DiegoPino** [comment]
@manez will merge when Travis passes. thanks

File: README.md [hunk 1]
```diff
@@ -4,6 +4,14 @@
 
 The Islandora Compound Object Solution Pack enables generic parent-child relationships between objects. The object view of a compound object is replaced by the view of its first child object. The included "Islandora Compound Object Navigation" block provides a thumbnail navigation of an object's siblings. A "Compound" management tab allows for the addition and removal of parent and child objects for each object.
 
+## Requirements
+
+This module requires the following modules/libraries:
+
+* [Islandora](https://github.com/islandora/islandora)
+* [JAIL](https://github.com/sebarmeli/JAIL) library
```
Comment (DiegoPino) [inline README.md]: Forgot about this for the committers call, but I guess any other @Islandora/7-x-1-x-committers could help with this. JAIL  is documented as a requirement but the module itself does not impose it as a requirement (only needed if enabled in the block). I personally vote for consistency across our docs but would also make sure people don't have to install libraries they won't be using. Any thoughts?

**Comment by rosiel** [comment]
at this point I am for including a list of conditional requirements after the requirements and explain which functionality it is a requirement for.

**Comment by manez** [comment]
How about:

## Requirements

This module requires the following modules/libraries:

* [Islandora](https://github.com/islandora/islandora)

This module has the following as optional requirements for certain features:

For Islandora Compound Object JAIL Display:

* [JAIL](https://github.com/sebarmeli/JAIL) library

**Comment by DiegoPino** [comment]
@manez there seems to be agreement on the wording, feel free to update whenever you want and will merge. Thanks!

**Comment by manez** [comment]
@DiegoPino updated with the new wording!

**Review by DiegoPino — APPROVED** [review]

**Pull request merged**